sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_replica_client.py

```diff
@@ -0,0 +1,310 @@
+    def refresh_feature_flags(
+        self,
+        refresh_on: Mapping[Tuple[str, str], Optional[str]],
+        feature_flag_selectors: List[SettingSelector],
+        headers: Dict[str, str],
+        **kwargs
+    ) -> Tuple[bool, Optional[Dict[Tuple[str, str], str]], Optional[List[Any]]]:
+        """
+        Gets the refreshed feature flags if they have changed.
+
+        :param Mapping[Tuple[str, str], Optional[str]] refresh_on: The feature flags to check for changes
+        :param List[SettingSelector] feature_flag_selectors: The selectors to use to load feature flags
+        :param Mapping[str, str] headers: The headers to use for the request
+
+        :return: A tuple with the first item being true/false if a change is detected. The second item is the updated
+        value of the feature flag sentinel keys. The third item is the updated feature flags.
+        :rtype: Tuple[bool, Union[Dict[Tuple[str, str], str], None], Union[List[Dict[str, str]], None]]
+        """
+        feature_flag_sentinel_keys: Mapping[Tuple[str, str], Optional[str]] = dict(refresh_on)
+        for (key, label), etag in feature_flag_sentinel_keys.items():
+            changed = self._check_configuration_setting(key=key, label=label, etag=etag, headers=headers, **kwargs)
+            if changed:
+                feature_flags, feature_flag_sentinel_keys = self.load_feature_flags(
+                    feature_flag_selectors, True, **kwargs
+                )
+                return True, feature_flag_sentinel_keys, feature_flags
+        return False, None, None
+
+    @distributed_trace
+    def get_configuration_setting(self, key: str, label: str, **kwargs) -> ConfigurationSetting:
+        """
+        Gets a configuration setting from the replica client.
+
+        :param str key: The key of the configuration setting
+        :param str label: The label of the configuration setting
+        :return: The configuration setting
+        :rtype: ConfigurationSetting
+        """
+        return self._client.get_configuration_setting(key=key, label=label, **kwargs)
+
+    def close(self) -> None:
+        """
+        Closes the connection to Azure App Configuration.
+        """
+        self._client.close()
+
+    def __enter__(self):
+        self._client.__enter__()
+        return self
+
+    def __exit__(self, *args):
+        self._client.__exit__(*args)
+
+
+class ReplicaClientManager:
+    def __init__(self, min_backoff_sec: int, max_backoff_sec: int):
+        self._replica_clients: List[ReplicaClient] = []
+        self._min_backoff_sec = min_backoff_sec
+        self._max_backoff_sec = max_backoff_sec
+
+    def set_clients(self, replica_clients: List[ReplicaClient]):
+        self._replica_clients.clear()
+        self._replica_clients.extend(replica_clients)
+
+    def get_active_clients(self):
+        active_clients = []
+        for client in self._replica_clients:
+            if client.backoff_end_time <= (time.time() * 1000):
```

Comment:
nit: Should whether or not a client is active be logic inside the ReplicaClient? Perhaps, a method called, `is_active()`